STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Exceptions.cs

```diff
@@ -0,0 +1,118 @@
+// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions;
+using Xeptions;
+
+namespace STX.REST.RESTFulSense.Clients.Services.Foundations.HttpExchanges
+{
+    internal partial class HttpExchangeService
+    {
+        private delegate ValueTask<HttpExchange> ReturningHttpExchangeFunction();
+
+        private static async ValueTask<HttpExchange> TryCatch(
+            ReturningHttpExchangeFunction returningHttpExchangeFunction)
+        {
+            try
+            {
+                return await returningHttpExchangeFunction();
+            }
+            catch (NullHttpExchangeException nullHttpExchangeException)
+            {
+                throw CreateHttpExchangeValidationException(
+                    exception: nullHttpExchangeException);
+            }
+            catch (InvalidHttpExchangeRequestException invalidHttpExchangeRequestException)
+            {
+                throw CreateHttpExchangeValidationException(
+                    invalidHttpExchangeRequestException);
+            }
+            catch (InvalidHttpExchangeHeaderException invalidHttpExchangeHeaderException)
+            {
+                throw CreateHttpExchangeValidationException(
+                    invalidHttpExchangeHeaderException);
+            }
+            catch(HttpRequestException httpRequestException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
+                        message: "Failed http request error occurred, contact support.",
+                        innerException: httpRequestException);
+
+                throw CreateHttpExchangeDependencyException(
+                    failedHttpExchangeException);
+            }
+            catch(TaskCanceledException taskCanceledException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
+                        message: "Request timeout error occurred, please contact support.",
+                        innerException: taskCanceledException);
+
+                throw CreateHttpExchangeDependencyException(
+                    failedHttpExchangeException);
+            }
+            catch(ObjectDisposedException objectDisposedException)
+            {
+                var failedHttpExchangeException =
+                    new FailedHttpExchangeException(
```

Comment:
Can we map this to local exception `ObjectDisposedHttpExchangeException`?

Reply:
Resolved